craft_parts/state_manager/stage_state.py

```diff
@@ -16,14 +16,25 @@
 
 """State definitions for the stage step."""
 
-from typing import Any, Dict
+from typing import Any, Dict, Optional
+
+from pydantic import validator
 
 from .step_state import StepState
 
 
 class StageState(StepState):
     """Context information for the stage step."""
 
+    overlay_hash: Optional[str] = None
+
+    @validator("overlay_hash")
+    @classmethod
+    def _validate_hex_string(cls, value):
```

Comment:
there's one exactly like this one defined also above...

Reply:
Reusing the validator.